### PR TITLE
Make the subcommand docstring test faster

### PR DESCRIPTION
**Pull request by bryant1410**
See https://github.com/allenai/allennlp/pull/3172#issuecomment-523664833 and #3183

### CODE REVIEW

**Comment by brendan-ai2** [comment]
Thanks for the fix, but we're actually removing this test. :(

**Comment by schmmd** [comment]
See https://github.com/allenai/allennlp/pull/3182 (it's disabled there)

**Comment by bryant1410** [comment]
@brendan-ai2 @schmmd @joelgrus @DeNeutoy I have made this test much faster now. It doesn't spawn processes anymore, it just checks the argument parser.

Just to clarify, this test checks that the subcommand help information within the docstrings are okay and not outdated (e.g.,`train.py` module docstring is like the output of `allennlp train --help`).

I believe it's fast enough and not flaky now to keep it enabled. If you still wanna keep it disabled, I still recommend merging this version as it's faster.

**Comment by bryant1410** [comment]
[I also recommend having timeouts.](https://github.com/allenai/allennlp/pull/3172#issuecomment-523698925)

File: allennlp/tests/commands/docstring_help_test.py [hunk 1]
```diff
@@ -29,16 +41,8 @@ def test_docstring_help(self):
...
-                actual_output = subprocess.run(
-                    str_call_from_python_subcommand_help,
-                    check=True,
-                    shell=True,
-                    stdout=subprocess.PIPE,
-                ).stdout.decode()
+                subcommand = match.group(2)
+                actual_output = _subcommand_help_output(subcommand)
 
                 pytorch_pretrained_bert_warning = (
```
Comment (joelgrus) [inline allennlp/tests/commands/docstring_help_test.py]: is this whole pytorch_pretrained_bert_warning part still necessary after these changes?

Comment (bryant1410) [reply allennlp/tests/commands/docstring_help_test.py]: No, it's not. I'm gonna remove it.

**Pull request merged**